packages/mongo/src/lib/MongoProvider.ts

```diff
@@ -179,22 +182,27 @@ export class MongoProvider<StoredValue = unknown> extends JoshProvider<StoredVal
 	public async [Method.Ensure](payload: EnsurePayload<StoredValue>): Promise<EnsurePayload<StoredValue>> {
 		const { key } = payload;
 
-		if (!(await this.has({ key, method: Method.Has, data: false, path: [] })))
+		if (!(await this.has({ key, method: Method.Has, data: false, path: [] })).data)
 			await this.set({ key, value: payload.defaultValue, method: Method.Set, path: [] });
 
-		Reflect.set(payload, 'data', await this.get({ key, method: Method.Get, path: [] }));
+		payload.data = (await this.get({ key, method: Method.Get, path: [] })).data as StoredValue;
 
 		return payload;
 	}
 
 	public async [Method.Every](payload: EveryByHookPayload<StoredValue>): Promise<EveryByHookPayload<StoredValue>>;
 	public async [Method.Every](payload: EveryByValuePayload): Promise<EveryByValuePayload>;
 	public async [Method.Every](payload: EveryPayload<StoredValue>): Promise<EveryPayload<StoredValue>> {
+		if ((await this.size({ method: Method.Size, data: 0 })).data === 0) {
+			payload.data = true;
+
+			return payload;
+		}
 		if (isEveryByHookPayload(payload)) {
 			const { hook } = payload;
 
-			for (const value of (await this.values({ method: Method.Values, data: [] })).data) {
-				const everyValue = await hook(value);
+			for (const val of (await this.values({ method: Method.Values, data: [] })).data) {
+				const everyValue = await hook(val);
```

Comment:
Not sure why this was changed.
```suggestion
			for (const value of (await this.values({ method: Method.Values, data: [] })).data) {
				const everyValue = await hook(value);
```